### PR TITLE
Add login management models

### PR DESCRIPTION
**Pull request by llewelld**
Adds models to allow management of logins (usernames and passwords)
through the frontend.

### CODE REVIEW

**Comment by llewelld** [comment]
This PR adds the qml needed to test the model:
https://github.com/sailfishos/sailfish-browser/pull/834/files

File: apps/browser/settings/logininfo.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    m_username = username;
+}
+
+void LoginInfo::setPassword(const QString &password)
+{
+    m_password = password;
+}
+
+bool LoginInfo::doLoginsMatch(const LoginInfo &other, bool ignorePassword, bool ignoreSchemes) const
+{
```
Comment (llewelld) [inline apps/browser/settings/logininfo.cpp:78]: There are indentation issues in this method; I'll fix them and push an update.

Comment (llewelld) [reply apps/browser/settings/logininfo.cpp:78]: Should now be fixed.

**Comment by llewelld** [comment]
Rebased and updated based on changes requested on #834.

**Comment by llewelld** [comment]
Updated to reflect changes made to https://git.sailfishos.org/mer-core/embedlite-components/merge_requests/126

**Comment by llewelld** [comment]
Rebased.

**Review by atatarov — APPROVED** [review]

**Review by mkenttala — APPROVED** [review]

**Pull request merged**